### PR TITLE
Dbx 77011/add polling for backup submitted 526s

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- Add a sidekiq job to poll the lighthouse benefits intake API for the status of form 526 submissions
- leverages the new  form 526 state to record results from the API
- adds a scope to the form 526 model for code readability
- schedules the job

## Related issue(s)

- [ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/77011)

## Testing done

- [ x ] *New code is covered by unit tests*
- previously we were manually polling these submissions and recording them in the state machine. This was relatively new, before that we simply weren't looking to see if things had failed.
- This job can be manually run in a production console safely, as data will only change in the way it's supposed to.

## What areas of the site does it impact?
Form 526 backup submission

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/form526_submission.rb
Comment (freeheeling) [inline app/models/form526_submission.rb:134]: Wouldn't `pending_backup_submissions` have a `backup_submitted_claim_id` that _is_ nil?

Comment (SamStuckey) [reply app/models/form526_submission.rb:134]: the state `delivered_to_backup` means that it's been sent (successfully) to LH.  HOWEVER, LH may still reject it later on, after their system has reviewed it.  This is definitely confusing, and is also the central reason we need polling.  SO the above code is really just saying "give me anything with a backup_submitted_claim_id that hasn't been finalized"

File: app/sidekiq/form526_status_polling_job.rb
Comment (SamStuckey) [reply app/sidekiq/form526_status_polling_job.rb:9]: yess that is a great catch.  this was copied from the other job and not updated.

Comment (SamStuckey) [reply app/sidekiq/form526_status_polling_job.rb:9]: ah sorry, if you look down at the `log_result` method, we are appending our usecase specific stuff to the stats key.  This is just the base.  We actually need to use this base in order to add our numbers to the `all_form` sub set of this key.

File: app/models/form526_submission.rb
Comment (SamStuckey) [reply app/models/form526_submission.rb:134]: I could call it `delivered_to_backup_pending_approval`?

Comment (freeheeling) [reply app/models/form526_submission.rb:134]: Got it. I suppose I wasn't clear on what the state of `delivered_to_backup` implied. It makes sense now that you've explained it and I referenced your earlier PR. If "backup" is LH, and the `backup_submitted _claim_id` is a unique ID received from LH acknowledging receipt of a backup submitted claim, then the given name would work. Or, if not, and wanting to keep it concise, a name like `pending_backup_finalization`, perhaps. I'll defer to you though as I feel you're better with words.

Comment (SamStuckey) [reply app/models/form526_submission.rb:134]: I think we are going to get an itteration on these states in the near future anyway for two reasons
- We need to add two more for remediation stuff
- these aren't really "states" as much as they are "end of life tags" or "remediation notes"
I'm ok with leaving it for now; but for the record [my next TODO has an action item to clean up / document what these states all mean](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/80624)

**Review by tblackwe — APPROVED** [review]

**Review by freeheeling — APPROVED** [review]
LGTM!

**Comment by rmtolmach** [comment]
@SamStuckey you'll need to update the codeowners for `spec/sidekiq/form526_status_polling_job_spec.rb` as well

File: .github/CODEOWNERS
Comment (freeheeling) [inline .github/CODEOWNERS:1213]: I would've placed this on line 1240 for alphabetical consistency, but this works.

**Review by freeheeling — APPROVED** [review]
Approved with a suggestion.

File: .github/CODEOWNERS
Comment (SamStuckey) [reply .github/CODEOWNERS:1213]: updated : )

**Review by freeheeling — APPROVED** [review]
🚢

**Comment by SamStuckey** [comment]
@rmtolmach ready for re-review when you can please

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**
